## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 502346`
Commit: [`b40406c48b92bad5e5b221886714b4e2cf358200`](https://github.com/NixOS/nixpkgs/commit/b40406c48b92bad5e5b221886714b4e2cf358200) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/b40406c48b92bad5e5b221886714b4e2cf358200..pull/502346/head))
Merge: [`d950efca7e018a67ef95e4b97c710a8037bc1b6c`](https://github.com/NixOS/nixpkgs/commit/d950efca7e018a67ef95e4b97c710a8037bc1b6c)

Logs: https://github.com/JamieMagee/nixpkgs-review-gha/actions/runs/23416181504


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>tests.home-assistant-component-tests.liebherr</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.pyliebherrhomeapi</li>
    <li>python313Packages.pyliebherrhomeapi.dist</li>
    <li>python314Packages.pyliebherrhomeapi</li>
    <li>python314Packages.pyliebherrhomeapi.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>tests.home-assistant-component-tests.liebherr</summary>
<pre>2026-03-23 00:27:12.480 INFO     MainThread homeassistant.setup:setup.py:393 Setting up select
2026-03-23 00:27:12.481 INFO     MainThread homeassistant.components.select:entity_platform.py:440 Setting up liebherr.select
2026-03-23 00:27:12.482 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new select.liebherr entity: select.test_fridge_bottom_zone_icemaker
2026-03-23 00:27:12.484 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new select.liebherr entity: select.test_fridge_top_zone_hydrobreeze
2026-03-23 00:27:12.484 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new select.liebherr entity: select.test_fridge_top_zone_biofresh_plus
=============================== warnings summary ===============================
homeassistant/runner.py:176
  /build/source/homeassistant/runner.py:176: DeprecationWarning: &#x27;asyncio.DefaultEventLoopPolicy&#x27; is deprecated and slated for removal in Python 3.16
    class HassEventLoopPolicy(asyncio.DefaultEventLoopPolicy):  # type: ignore[name-defined,misc]

tests/conftest.py:155
  /build/source/tests/conftest.py:155: DeprecationWarning: &#x27;asyncio.set_event_loop_policy&#x27; is deprecated and slated for removal in Python 3.16
    asyncio.set_event_loop_policy(runner.HassEventLoopPolicy(False))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--------------------------- snapshot report summary ----------------------------
33 snapshots passed.
=========================== short test summary info ============================
FAILED tests/components/liebherr/test_select.py::test_select_current_option_none_mode - dataclasses.FrozenInstanceError: cannot assign to field &#x27;current_mode&#x27;
=================== 1 failed, 57 passed, 2 warnings in 5.50s ===================</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>tests.home-assistant-component-tests.liebherr</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.pyliebherrhomeapi</li>
    <li>python313Packages.pyliebherrhomeapi.dist</li>
    <li>python314Packages.pyliebherrhomeapi</li>
    <li>python314Packages.pyliebherrhomeapi.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>tests.home-assistant-component-tests.liebherr</summary>
<pre>2026-03-23 00:26:21.614 INFO     MainThread homeassistant.setup:setup.py:393 Setting up select
2026-03-23 00:26:21.615 INFO     MainThread homeassistant.components.select:entity_platform.py:440 Setting up liebherr.select
2026-03-23 00:26:21.616 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new select.liebherr entity: select.test_fridge_bottom_zone_icemaker
2026-03-23 00:26:21.616 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new select.liebherr entity: select.test_fridge_top_zone_hydrobreeze
2026-03-23 00:26:21.617 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new select.liebherr entity: select.test_fridge_top_zone_biofresh_plus
=============================== warnings summary ===============================
homeassistant/runner.py:176
  /build/source/homeassistant/runner.py:176: DeprecationWarning: &#x27;asyncio.DefaultEventLoopPolicy&#x27; is deprecated and slated for removal in Python 3.16
    class HassEventLoopPolicy(asyncio.DefaultEventLoopPolicy):  # type: ignore[name-defined,misc]

tests/conftest.py:155
  /build/source/tests/conftest.py:155: DeprecationWarning: &#x27;asyncio.set_event_loop_policy&#x27; is deprecated and slated for removal in Python 3.16
    asyncio.set_event_loop_policy(runner.HassEventLoopPolicy(False))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--------------------------- snapshot report summary ----------------------------
33 snapshots passed.
=========================== short test summary info ============================
FAILED tests/components/liebherr/test_select.py::test_select_current_option_none_mode - dataclasses.FrozenInstanceError: cannot assign to field &#x27;current_mode&#x27;
=================== 1 failed, 57 passed, 2 warnings in 4.37s ===================</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.pyliebherrhomeapi</li>
    <li>python313Packages.pyliebherrhomeapi.dist</li>
    <li>python314Packages.pyliebherrhomeapi</li>
    <li>python314Packages.pyliebherrhomeapi.dist</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.pyliebherrhomeapi</li>
    <li>python313Packages.pyliebherrhomeapi.dist</li>
    <li>python314Packages.pyliebherrhomeapi</li>
    <li>python314Packages.pyliebherrhomeapi.dist</li>
  </ul>
</details>

--- a/pkgs/development/python-modules/pyliebherrhomeapi/default.nix
+++ b/pkgs/development/python-modules/pyliebherrhomeapi/default.nix
@@ -1,0 +1,43 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  aiohttp,
+  pytestCheckHook,
+  pytest-asyncio,
+  pytest-timeout,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "pyliebherrhomeapi";
+  version = "0.3.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "mettolen";
+    repo = "pyliebherrhomeapi";
+    tag = finalAttrs.version;
+    hash = "sha256-J+s9oRKK2DWZV05M1bvDEBqczaJXuif4t3eOJUgrAu0=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [ aiohttp ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-asyncio
+    pytest-timeout
+  ];
+
+  pythonImportsCheck = [ "pyliebherrhomeapi" ];
+
+  meta = {
+    description = "Python library for Liebherr Home API";
+    homepage = "https://github.com/mettolen/pyliebherrhomeapi";
+    changelog = "https://github.com/mettolen/pyliebherrhomeapi/releases/tag/${finalAttrs.version}";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.jamiemagee ];
+  };
+})

--- a/pkgs/servers/home-assistant/component-packages.nix
+++ b/pkgs/servers/home-assistant/component-packages.nix
@@ -3421,7 +3421,8 @@
       ];
     "liebherr" =
       ps: with ps; [
-      ]; # missing inputs: pyliebherrhomeapi
+        pyliebherrhomeapi
+      ];
     "life360" =
       ps: with ps; [
       ];
@@ -7709,6 +7710,7 @@
     "lg_thinq"
     "libre_hardware_monitor"
     "lidarr"
+    "liebherr"
     "life360"
     "lifx"
     "light"

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -14094,6 +14094,8 @@ self: super: with self; {
 
   pylibsrtp = callPackage ../development/python-modules/pylibsrtp { };
 
+  pyliebherrhomeapi = callPackage ../development/python-modules/pyliebherrhomeapi { };
+
   pylink-square = callPackage ../development/python-modules/pylink-square { };
 
   pylint = callPackage ../development/python-modules/pylint { };

